### PR TITLE
Add tutorials

### PR DESCRIPTION
**Pull request by kentsanggds**
## What

Add tutorials to help developers familiarise themselves with developing on the kubernetes cluster.

## Reference

https://trello.com/c/X357aij8/988-write-up-codelab-sessions

### CODE REVIEW

File: source/get-started/tutorials/app-config-deploy-helm-chart/index.html.md.erb [hunk 1]
```diff
@@ -0,0 +1,32 @@
+---
+title: App configuration and deployment using the govuk helm chart
+weight: 20
+last_reviewed_on: 2022-09-08
+review_in: 6 months
+---
+
+# App configuration and deployment using the govuk helm chart
```
Comment (theseanything) [inline source/get-started/tutorials/app-config-deploy-helm-chart/index.html.md.erb:8]: Nit: if you change the title, should update this too.

File: source/get-started/tutorials/app-config-deploy-helm-chart/index.html.md.erb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+This tutorial aims to guide you through the process of updating configuration in the govuk-helm-charts repo for the test app and seeing the deployment following the changes you made.
+
+In a real world situation you might also update the number of CPUs for the app, memory size etc.
+
+1. App configuration update in GOVUK helm chart
```
Comment (theseanything) [inline source/get-started/tutorials/app-config-deploy-helm-chart/index.html.md.erb:14]: These step by steps should be instructions e.g. start with a verb. Also changed govuk-helm-chart for consistency.
```suggestion
1. Update the application configuration in the govuk-helm-chart repository
```

File: source/get-started/tutorials/app-config-deploy-helm-chart/index.html.md.erb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+Add your `ENV_MESSAGE_<your name>` environment variable value for the test app configuration in [govuk-helm-charts](https://github.com/alphagov/govuk-helm-charts/blob/main/charts/argocd-apps/values-integration.yaml#L998) in a new branch.
+
+2. Deploying the changes to the app
+
+Create the pull request on your branch and post it for review on the #govuk-replatforming-class channel. After your PR has been approved for merging into the main branch, merge the PR in and look at the [argo CD apps workflow](https://argo.eks.integration.govuk.digital/applications/argocd-apps?view=tree&resource=) as your change gets deployed.
```
Comment (theseanything) [inline source/get-started/tutorials/app-config-deploy-helm-chart/index.html.md.erb:20]: Nit: simplification of wording
```suggestion
Create the pull request (PR) for your branch and post it for review on the #govuk-replatforming-class channel. After your PR has been approved and passed status checks, you can merge your PR.
```
Also slightly confusing that we tell them to go Argo, when the next step is also to go to Argo.

File: source/get-started/tutorials/app-config-deploy-helm-chart/index.html.md.erb [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+3. See the changes in Argo
+
+You should be able to see your changes in the manifest within Argo (https://argo.eks.integration.govuk.digital/applications/argocd-apps?view=tree&resource=&node=argoproj.io%2FApplication%2Fcluster-services%2Fgovuk-replatform-test-app%2F0)
```
Comment (theseanything) [inline source/get-started/tutorials/app-config-deploy-helm-chart/index.html.md.erb:24]: Do we wanna show them the change in the actual deployment resource? Or link the actual Argo Application for the test app, rather than the argocd-app (which is just a app of apps)?

File: source/get-started/tutorials/app-config-deploy-helm-chart/index.html.md.erb [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+4. See your environment message in the test app
+
+Navigate to https://govuk-replatform-test-app.eks.integration.govuk.digital/?status=200 to see the `ENV_MESSAGE` as part of the page output, you have now successfully deployed the change to the GOV.UK kubernetes integration cluster.
```
Comment (theseanything) [inline source/get-started/tutorials/app-config-deploy-helm-chart/index.html.md.erb:28]:
```suggestion
Navigate to https://govuk-replatform-test-app.eks.integration.govuk.digital/?status=200 to see the `ENV_MESSAGE` as part of the page output, you have now successfully deployed the change to the GOV.UK Kubernetes integration cluster.
```
not sure if our environments also use capitals?

File: source/get-started/tutorials/app-config-deploy-helm-chart/index.html.md.erb [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+2. Deploying the changes to the app
+
+Create the pull request on your branch and post it for review on the #govuk-replatforming-class channel. After your PR has been approved for merging into the main branch, merge the PR in and look at the [argo CD apps workflow](https://argo.eks.integration.govuk.digital/applications/argocd-apps?view=tree&resource=) as your change gets deployed.
+
+3. See the changes in Argo
+
+You should be able to see your changes in the manifest within Argo (https://argo.eks.integration.govuk.digital/applications/argocd-apps?view=tree&resource=&node=argoproj.io%2FApplication%2Fcluster-services%2Fgovuk-replatform-test-app%2F0)
+
+4. See your environment message in the test app
```
Comment (theseanything) [inline source/get-started/tutorials/app-config-deploy-helm-chart/index.html.md.erb:26]:
```suggestion
4. See your message in the test app
```
"environment message" doesn't really mean anything? it's either environment variable or a message?

File: source/get-started/tutorials/app-config-deploy-helm-chart/index.html.md.erb [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+Navigate to https://govuk-replatform-test-app.eks.integration.govuk.digital/?status=200 to see the `ENV_MESSAGE` as part of the page output, you have now successfully deployed the change to the GOV.UK kubernetes integration cluster.
+
+5. Tidy up your `ENV_MESSAGE`
+
+Please remove your `ENV_MESSAGE` once you are happy that you have seen your changes in the test app manifest within Argo and on the test app website.
```
Comment (theseanything) [inline source/get-started/tutorials/app-config-deploy-helm-chart/index.html.md.erb:32]: Maybe with suggesting to do git revert?

**Review by theseanything — COMMENTED** [review]
Sorry Ken - I've only got around to look at the first tutorial so far

File: source/get-started/tutorials/app-config-deploy-helm-chart/index.html.md.erb [hunk 8]
```diff
@@ -0,0 +1,32 @@
+---
+title: App configuration and deployment using the govuk helm chart
```
[the remark this answers is not in the document]
Comment (kentsanggds) [reply source/get-started/tutorials/app-config-deploy-helm-chart/index.html.md.erb:2]: OK, I'm just wondering if we need to make a distinction between making config changes on the app and making config changes on the app within the govuk helm chart repo? as not all app config change are in the helm chart?

File: source/get-started/tutorials/app-config-deploy-helm-chart/index.html.md.erb [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+weight: 20
+last_reviewed_on: 2022-09-08
+review_in: 6 months
+---
+
+# App configuration and deployment using the govuk helm chart
+
+This tutorial aims to guide you through the process of updating configuration in the govuk-helm-charts repo for the test app and seeing the deployment following the changes you made.
```
[the remark this answers is not in the document]
Comment (kentsanggds) [reply source/get-started/tutorials/app-config-deploy-helm-chart/index.html.md.erb:10]: I'll reword it to make it clearer

File: source/get-started/tutorials/app-config-deploy-helm-chart/index.html.md.erb [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+
+In a real world situation you might also update the number of CPUs for the app, memory size etc.
```
[the remark this answers is not in the document]
Comment (kentsanggds) [reply source/get-started/tutorials/app-config-deploy-helm-chart/index.html.md.erb:12]: Yes, as the changes being made in the helm chart are just related to environment variables

File: source/get-started/tutorials/app-config-deploy-helm-chart/index.html.md.erb (hunk 2 above)
Comment (kentsanggds) [reply source/get-started/tutorials/app-config-deploy-helm-chart/index.html.md.erb:14]: OK, will update

File: source/get-started/tutorials/app-config-deploy-helm-chart/index.html.md.erb (hunk 4 above)
Comment (kentsanggds) [reply source/get-started/tutorials/app-config-deploy-helm-chart/index.html.md.erb:24]: Probably the Argo Application, will update the link

File: source/get-started/tutorials/app-config-deploy-helm-chart/index.html.md.erb (hunk 5 above)
Comment (kentsanggds) [reply source/get-started/tutorials/app-config-deploy-helm-chart/index.html.md.erb:28]: No idea, I'll have a poke around

File: source/get-started/tutorials/app-config-deploy-helm-chart/index.html.md.erb (hunk 6 above)
Comment (kentsanggds) [reply source/get-started/tutorials/app-config-deploy-helm-chart/index.html.md.erb:26]: Wanted to make a distinction between the message set up in the helm chart and the message dropped in the test app which is why I've added `environment` here.

File: source/get-started/tutorials/app-config-deploy-helm-chart/index.html.md.erb (hunk 7 above)
Comment (kentsanggds) [reply source/get-started/tutorials/app-config-deploy-helm-chart/index.html.md.erb:32]: Was thinking about what happens when several people are doing a codelab session, will it cause more confusion with git reverts? I guess in the end it will be cleaner to do it this way as all changes introduced will be removed, they just need to let people know.

File: source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+review_in: 6 months
+---
+
+# App update, deployment, monitoring and logs
+
+This tutorial will take you through steps to make changes to the test app which will then get deployed to the cluster. Once deployed you should be able to see your changes in the logs and make requests to return specified status codes which should be visible in the grafana dashboard. 
+
+1. App update in the test app
+
+Add a new text file with a prefix of your name, e.g. jblogs-test.txt, to the [test app](https://github.com/alphagov/govuk-replatform-test-app/tree/main/messages) `messages` directory on the `main` branch, this can be done via your chosen IDE or directly on github.
```
Comment (theseanything) [inline source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb:14]: Think it would be good to give more context before asking users to immediately make a new text file. What is the test app, where is it located, briefly how does it work?

How come we're making changes directly on main instead of raising a PR? Think it'd be good to have as realistic development process as other GOV.UK apps. Also it'd give them a change to see CI running (if we add it later).

To be more concise, don't think we need to say this - developer are aware of how to make changes. 
```suggestion
Add a new text file with a prefix of your name, e.g. jblogs-test.txt, to the [test app](https://github.com/alphagov/govuk-replatform-test-app/tree/main/messages) `messages` directory on the `main` branch.
```

File: source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+
+2. Notify other tutorial devs
+
+Send a slack message to the #govuk-replatforming-class channel to let them know that you are planning to deploy changes to the test app as this will cause them to require a pull of the latest main branch.
```
Comment (theseanything) [inline source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb:18]: As above - not sure why we're committing directly to main?

File: source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+
+3. Deploying the changes to the app
```
Comment (theseanything) [inline source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb:20]: This doesn't make sense - you don't deploy a change "to an app"? You can deploy an app to an environment.
```suggestion
3. Deploying the changes to the app
```

File: source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+
+Commit your changes to the main branch to start the deployment process. 
+
+You would normally create a PR before pushing changes to the main branch but as you will be just adding text files to a directory there should be little chance of anything breaking.
+
+4. Seeing the app changes being deployed
+
+You can monitor the app changes via [github actions](https://github.com/alphagov/govuk-replatform-test-app/actions) which triggers a [job](https://argo-workflows.eks.integration.govuk.digital/workflows/apps?limit=500) to update the image tag stored in [govuk-helms-chart](https://github.com/alphagov/govuk-helm-charts/blob/main/charts/argocd-apps/image-tags/integration/govuk-replatform-test-app) repo.
```
Comment (theseanything) [inline source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb:28]:
```suggestion
You check the ["Deploy" GitHub Action workflow](https://github.com/alphagov/govuk-replatform-test-app/actions) to see the deployment has begun. This workflow builds a new container image with your application changes, and then triggers an [Argo workflow](https://argo-workflows.eks.integration.govuk.digital/workflows/apps?limit=500) to update the image tag stored in [govuk-helms-chart](https://github.com/alphagov/govuk-helm-charts/blob/main/charts/argocd-apps/image-tags/integration/govuk-replatform-test-app) repository. 
```

You don't really monitor "app changes" from GitHub Actions, nor do you monitor the whole deployment.

File: source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb [hunk 15]
```diff
@@ -0,0 +1,64 @@
...
+
+Commit your changes to the main branch to start the deployment process. 
+
+You would normally create a PR before pushing changes to the main branch but as you will be just adding text files to a directory there should be little chance of anything breaking.
+
+4. Seeing the app changes being deployed
+
+You can monitor the app changes via [github actions](https://github.com/alphagov/govuk-replatform-test-app/actions) which triggers a [job](https://argo-workflows.eks.integration.govuk.digital/workflows/apps?limit=500) to update the image tag stored in [govuk-helms-chart](https://github.com/alphagov/govuk-helm-charts/blob/main/charts/argocd-apps/image-tags/integration/govuk-replatform-test-app) repo. 
+
+This causes the [app in the cluster](https://argo.eks.integration.govuk.digital/applications/argocd-apps?view=tree&resource=&node=argoproj.io%2FApplication%2Fcluster-services%2Fgovuk-replatform-test-app%2F0) to be out of sync with the image tag on the govuk-helms-chart so argo will attempt to redeploy the app to keep it in sync.
```
Comment (theseanything) [inline source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb:30]: Check consistency with nouns and capitalisation e.g. argo -> Argo

File: source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb [hunk 16]
```diff
@@ -0,0 +1,64 @@
+---
+title: App update, deployment, monitoring and logs
```
[the remark this answers is not in the document]
Comment (kentsanggds) [reply source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb:2]: ok, I'll make it clearer, but isn't it monitor metrics from the app rather than the deployment?

File: source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb (hunk 11 above)
Comment (kentsanggds) [reply source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb:14]: Hmmm, thought that creating a PR for review would create more noise on the channel? and since more of us are familiar with creating PRs and merging them in after review? maybe we can skip the review part and just get them to merge it in though. I'll update the text

File: source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb (hunk 12 above)
Comment (kentsanggds) [reply source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb:18]: Will reword

File: source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb (hunk 13 above)
Comment (kentsanggds) [reply source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb:20]: I'll reword

File: source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb (hunk 14 above)
Comment (kentsanggds) [reply source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb:28]: I'll reword

File: source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb (hunk 15 above)
Comment (kentsanggds) [reply source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb:30]: OK

File: source/get-started/tutorials/app-config-deploy-helm-chart/index.html.md.erb (hunk 8 above)
[the remark this answers is not in the document]
Comment (kentsanggds) [reply source/get-started/tutorials/app-config-deploy-helm-chart/index.html.md.erb:2]: OK, I'll update it for this

File: source/get-started/tutorials/app-config-deploy-helm-chart/index.html.md.erb (hunk 1 above)
Comment (kentsanggds) [reply source/get-started/tutorials/app-config-deploy-helm-chart/index.html.md.erb:8]: ok

File: source/get-started/tutorials/app-config-deploy-helm-chart/index.html.md.erb [hunk 17]
```diff
@@ -0,0 +1,32 @@
...
+
+1. App configuration update in GOVUK helm chart
+
+Add your `ENV_MESSAGE_<your name>` environment variable value for the test app configuration in [govuk-helm-charts](https://github.com/alphagov/govuk-helm-charts/blob/main/charts/argocd-apps/values-integration.yaml#L998) in a new branch.
```
[the remark this answers is not in the document]
Comment (kentsanggds) [reply source/get-started/tutorials/app-config-deploy-helm-chart/index.html.md.erb:16]: ok

File: source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb [hunk 18]
```diff
@@ -0,0 +1,64 @@
...
+weight: 20
+last_reviewed_on: 2022-09-08
+review_in: 6 months
+---
+
+# App update, deployment, monitoring and logs
+
+This tutorial will take you through steps to make changes to the test app which will then get deployed to the cluster. Once deployed you should be able to see your changes in the logs and make requests to return specified status codes which should be visible in the grafana dashboard. 
+
+1. App update in the test app
```
[the remark this answers is not in the document]
Comment (kentsanggds) [reply source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb:12]: ok

File: source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb [hunk 19]
```diff
@@ -0,0 +1,64 @@
+---
+title: App update, deployment, monitoring and logs
+weight: 20
+last_reviewed_on: 2022-09-08
+review_in: 6 months
+---
+
+# App update, deployment, monitoring and logs
+
+This tutorial will take you through steps to make changes to the test app which will then get deployed to the cluster. Once deployed you should be able to see your changes in the logs and make requests to return specified status codes which should be visible in the grafana dashboard.
```
[the remark this answers is not in the document]
Comment (kentsanggds) [reply source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb:10]: ok

File: source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb (hunk 12 above)
Comment (theseanything) [reply source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb]: Ah sorry this wasn't a wording issue, was wondering why we are committing directly to main, instead of raising a PR?

Comment (theseanything) [reply source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb]: Sorry you talked about it in an earlier comment above.

File: source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb (hunk 11 above)
Comment (theseanything) [reply source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb]: Are you worried about noise in terms people asking for PR reviews? If its just the test app repo - you could switch off the requirement to need a person to review, and allow people to merge their PR straight away. That way people don't need to ask for reviews, nor do they need to worry about committing directly to main and messing up someone else's work.

File: source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb (hunk 16 above)
[the remark this answers is not in the document]
Comment (theseanything) [reply source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb]: Sorry didn't see your question as you'd resolved this comment thread. Sure! I was just guess as to what you meant by "App update, deployment, monitoring and logs"

File: source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb [hunk 20]
```diff
@@ -0,0 +1,66 @@
...
+
+This tutorial will teach you how to deploy changes to an app and how to view application metrics and logs.
+
+In this tutorial, you will make a change to the example test app to print your message in the application output and view the message in the application logs. Later, you will make HTTP requests to the example test app and simulate responses with different status codes and view those response metics in Grafana. 
+
+1. Update the example test app with your message
+
+- Clone the [example test app](https://github.com/alphagov/govuk-replatform-test-app/tree/main/messages).
+- Create a new branch to add your changes.
+- Add a new text file with a prefix of your name, e.g. jblogs-test.txt, to the [test app](https://github.com/alphagov/govuk-replatform-test-app/tree/main/messages) `messages` directory.
```
Comment (theseanything) [inline source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb:18]: Nit: 
```suggestion
- Add a new text file with a prefix of your name, e.g. jblogs-test.txt, to the [test app](https://github.com/alphagov/govuk-replatform-test-app/tree/main/messages) in the `messages` directory.
```

File: source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb [hunk 21]
```diff
@@ -0,0 +1,66 @@
...
+
+2. Notify other tutorial developers
+
+Send a slack message to the #govuk-replatforming-class channel to let them know that you are planning to deploy changes to the example test app as this will cause them to require a pull of the latest main branch.
+
+3. Deploy the app changes to the integration environment
+
+Create a pull request (PR) and then merge your change in to kick off the deployment process.
+
```
Comment (theseanything) [inline source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb:27]: This doesn't make sense. Step 2 assumes you've committed to main, Step 3 assumes you're making a PR?

File: source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb [hunk 22]
```diff
@@ -0,0 +1,66 @@
...
+4. Check that the app changes have started the deployment
```
Comment (theseanything) [inline source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb:28]: Nit: v minor - but feel like the use of the word check might imply that this is something developers will actually need to do when the deploy an app when doing real development work. We are getting them to check as part of this tutorial, so they understand what's going on. But they don't actually need to worry about this when deploying. 

```suggestion
4. See that the app changes have started the deployment
```

File: source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb [hunk 23]
```diff
@@ -0,0 +1,66 @@
...
+
+You check the ["Deploy" GitHub Action workflow](https://github.com/alphagov/govuk-replatform-test-app/actions) to see that the deployment has begun. This workflow builds a new container image with your application changes, and then triggers an [Argo workflow](https://argo-workflows.eks.integration.govuk.digital/workflows/apps?limit=500) to update the image tag stored in [govuk-helms-chart](https://github.com/alphagov/govuk-helm-charts/blob/main/charts/argocd-apps/image-tags/integration/govuk-replatform-test-app) repository.
```
Comment (theseanything) [inline source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb:30]:
```suggestion
You can check the ["Deploy" GitHub Action workflow](https://github.com/alphagov/govuk-replatform-test-app/actions) to see that the deployment has begun. This workflow builds a new container image with your application changes, and then triggers an [Argo workflow](https://argo-workflows.eks.integration.govuk.digital/workflows/apps?limit=500) to update the image tag stored in [govuk-helms-chart](https://github.com/alphagov/govuk-helm-charts/blob/main/charts/argocd-apps/image-tags/integration/govuk-replatform-test-app) repository.  
```

File: source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb [hunk 24]
```diff
@@ -0,0 +1,66 @@
...
+
+You check the ["Deploy" GitHub Action workflow](https://github.com/alphagov/govuk-replatform-test-app/actions) to see that the deployment has begun. This workflow builds a new container image with your application changes, and then triggers an [Argo workflow](https://argo-workflows.eks.integration.govuk.digital/workflows/apps?limit=500) to update the image tag stored in [govuk-helms-chart](https://github.com/alphagov/govuk-helm-charts/blob/main/charts/argocd-apps/image-tags/integration/govuk-replatform-test-app) repository.  
+
+This causes the [app in the cluster](https://argo.eks.integration.govuk.digital/applications/argocd-apps?view=tree&resource=&node=argoproj.io%2FApplication%2Fcluster-services%2Fgovuk-replatform-test-app%2F0) to be out of sync with the image tag on the govuk-helms-chart so Argo will attempt to redeploy the app to keep it in sync.
+
+5. Check your message on the example test app website
+
+To see your message on the example test app website visit the [example test app](https://govuk-replatform-test-app.eks.integration.govuk.digital/?status=200).
+
+6. View app metrics on the grafana dashboard
```
Comment (theseanything) [inline source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb:38]: Capitalisation for proper nouns. Again you check for consistency throughout this document?

```suggestion
6. View app metrics on the Grafana dashboard
```

File: source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb [hunk 25]
```diff
@@ -0,0 +1,66 @@
...
+
+Change the status parameter in the test app url to a 5xx status in order to see the error appearing in the [grafana dashboard](https://grafana.eks.integration.govuk.digital/d/000000109/app-request-rates-errors-durations?orgId=1&refresh=10s&var-namespace=tutorials&var-app=govuk-replatform-test-app&var-quantile=All&var-error_status=All).
+
+Be aware that if it is difficult to see your metrics in grafana because there are multiple users running the tutorial, it should be possible to view your metrics in [prometheus](https://prometheus.eks.integration.govuk.digital/graph?g0.expr=http_request_duration_seconds%7Bjob%3D%22apps%2Fgovuk-replatform-test-app%22%2C%20install_id%3D%221660837311%22%7D&g0.tab=0&g0.stacked=0&g0.show_exemplars=0&g0.range_input=1h) instead using a filter for your identifier.
```
Comment (theseanything) [inline source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb:42]: Do we want to expose this to users? I feel like we shouldn't concern people with Prometheus, as they should be using Grafana for metrics. Maybe we can leave this out - unless we know its an actual problem for people doing this tutorial?

File: source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb [hunk 26]
```diff
@@ -0,0 +1,66 @@
...
+
+Using the identifier - `1660815896` in this request -
+
+`https://kibana.logit.io/s/42f4d2d5-e9ce-451f-8ffc-cdb25bd624f8/app/discover#/?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:now-4d,to:now))&_a=(columns:!(_source),filters:!(),index:'8ac115c0-aac1-11e8-88ea-0383c11b333a',interval:auto,query:(language:kuery,query:"1660815896"),sort:!())` 
+
+Will bring up a page for logs relevant to your app deployment.
+
+8. Tidy up your messages
+
+Once you are happy that you have seen your requests in grafana and logs for your deployment in kibana please remove your messages.
```
Comment (theseanything) [inline source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb:62]:
```suggestion
Once you are happy that you have seen your requests in Grafana and logs for your deployment in Kibana please remove your messages.
```

File: source/get-started/tutorials/index.html.md.erb [hunk 27]
```diff
@@ -0,0 +1,14 @@
...
+review_in: 6 months
+---
+
+# Tutorials
+
+These are tutorials for developers to familiarise yourself with the new GOV.UK kubernetes cluster.
+
+- App configuration and deployment using the govuk helm chart
+
+- App update, deployment, monitoring and logs using a test app
```
Comment (theseanything) [inline source/get-started/tutorials/index.html.md.erb:14]: These need updating?

File: source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb (hunk 11 above)
Comment (kentsanggds) [reply source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb]: yes, I think I updated the text to create the PR and merge it in

File: source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb (hunk 21 above)
Comment (kentsanggds) [reply source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb:27]: If they create a PR and then merge it into main won't other codelab devs will need to update their branch before merging their PRs?

File: source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb (hunk 25 above)
Comment (kentsanggds) [reply source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb:42]: ok, lets see if this becomes an issue, I'll remove the line

File: source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb (hunk 26 above)
Comment (kentsanggds) [reply source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb:62]: ok

File: source/get-started/tutorials/index.html.md.erb (hunk 27 above)
Comment (kentsanggds) [reply source/get-started/tutorials/index.html.md.erb:14]: ok

File: source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb (hunk 16 above)
[the remark this answers is not in the document]
Comment (kentsanggds) [reply source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb]: ok

File: source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb (hunk 20 above)
Comment (kentsanggds) [reply source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb:18]: ok

File: source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb (hunk 22 above)
Comment (kentsanggds) [reply source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb:28]: ok

File: source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb (hunk 23 above)
Comment (kentsanggds) [reply source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb:30]: ok

File: source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb (hunk 24 above)
Comment (kentsanggds) [reply source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb:38]: ok

File: source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb (hunk 21 above)
Comment (theseanything) [reply source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb]: If you raise a PR, it should show you a diff against the latest main commit. It should just be a normal git workflow.

Comment (kentsanggds) [reply source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb]: ok, I can remove the line

File: source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb [hunk 28]
```diff
@@ -0,0 +1,60 @@
...
+
+Will bring up a page for logs relevant to your app deployment.
+
+7. Tidy up your messages
+
+Once you are happy that you have seen your requests in Grafana and logs for your deployment in Kibana please revert your PR to remove your messges.
+
+## Maintenance
+
+Do not merge changes to the base test app directly to the `main` branch. If you want to make changes to the test app please create a PR and send it to the `#replatforming` slack channel for review.
```
Comment (theseanything) [inline source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb:60]: How come we're still saying:
"Do not merge changes to the base test app directly to the `main` branch"?

Comment (kentsanggds) [reply source/get-started/tutorials/app-update-deploy-monitor-logs/index.html.md.erb:60]: good spot, I'll remove that line

**Review by theseanything — APPROVED** [review]
